docs/proposals/07-shoot-control-plane-migration.md

```diff
@@ -0,0 +1,236 @@
+# Shoot Control Plane Migration
+
+## Motivation
+
+Currently moving the control plane of a shoot cluster can only be done manually and requires deep knowledge of how exactly to transfer the resources and state from one seed to another. This can make it slow and prone to errors.
+
+Automatic migration can be very useful in a couple of scenarios:
+- Seed goes down and can't be repaired (fast enough or at all) and it's control planes need to be brought to another seed
+- Seed needs to be changed, but this operation requires the recreation of the seed (e.g. turn a single-AZ seed into a multi-AZ seed)
+- Seeds need to be rebalanced
+- New seeds became available in a region closer to/in the region of the workers and the control plane should be moved there to improve latency
+- Gardener ring, which is a self-supporting setup/underlay for a highly available (usually cross-region) Gardener deployment
+
+
+## Goals
+
+* Provide a mechanism to migrate the control plane of a shoot cluster from one seed to another
+* The mechanism should support migration from a seed which is no longer reachable (Disaster Recovery)
+* The mechanism can automatically decide whether a seed is unhealthy and control planes need to be migrated to a new seed.
```

Comment:
That's not a goal of this GEP/implementation - this topic is large in its own and will/must be done separately.